server/sonar-server/src/main/java/org/sonar/server/computation/measure/MeasureImpl.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.measure;
+
+import com.google.common.base.Preconditions;
+import java.util.Locale;
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public final class MeasureImpl implements Measure {
+
+  private final ValueType valueType;
+  @Nullable
+  private final Double value;
+  @Nullable
+  private final String data;
+  @Nullable
+  private AlertStatus alertStatus;
+  @Nullable
+  private String alertText;
+
+  protected MeasureImpl(ValueType valueType, @Nullable Double value, @Nullable String data) {
+    this.valueType = valueType;
+    this.value = value;
+    this.data = data;
+  }
+
+  public static MeasureImpl create(boolean value, @Nullable String data) {
+    return new MeasureImpl(ValueType.BOOLEAN, value ? 1.0d : 0.0d, data);
+  }
+
+  public static MeasureImpl create(int value, @Nullable String data) {
+    return new MeasureImpl(ValueType.INT, (double) value, data);
+  }
+
+  public static MeasureImpl create(long value, @Nullable String data) {
+    return new MeasureImpl(ValueType.LONG, (double) value, data);
+  }
+
+  public static MeasureImpl create(double value, @Nullable String data) {
+    return new MeasureImpl(ValueType.DOUBLE, value, data);
+  }
+
+  public static MeasureImpl create(String value) {
+    return new MeasureImpl(ValueType.STRING, null, checkNotNull(value));
+  }
+
+  public static MeasureImpl createNoValue() {
+    return new MeasureImpl(ValueType.NO_VALUE, null, null);
+  }
+
+  @Override
+  public ValueType getValueType() {
+    return valueType;
+  }
+
+  @Override
+  public boolean getBooleanValue() {
+    checkValueType(ValueType.BOOLEAN);
+    return value == 1.0d;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1244)